### ISSUE TITLE
Building project with MegaCore no longer works

### ISSUE THREAD

**Issue by GunZi200** [platformio/platform-atmelavr#251]
I have tried with both ATmega2560 and ATmega2561.

This has never been a problem before.

![image](https://user-images.githubusercontent.com/8218499/114211492-3b85f400-9950-11eb-92cc-878b73e1f724.png)

**Comment by GunZi200** [comment]
Here is my `.ini` file for one of my projects. It can be used to reproduce.

```ini
[env]
platform = atmelavr
framework = arduino
monitor_speed = 115200

board_build.f_cpu = 16000000L        ; 16MHz CPU frequency
board_hardware.uart = uart0          ; For Bootloader
board_hardware.oscillator = external ; External 16MHz crystal

upload_flags = -Pusb

build_flags =
    -Wunused-parameter ; Turn on compiler warning when functions have unused parameters

[env:megaatmega2561]
board = ATmega2561
board_build.mcu = atmega2561

[env:megaatmega2560]
board = ATmega2560
board_build.mcu = atmega2560
```

**Comment by GunZi200** [comment]
I can reproduce the problem:

`platform = atmelavr@3.2.0` does not work

`platform = atmelavr@3.1.0` works

**Comment by GunZi200** [comment]
If I revert line 78 here https://github.com/platformio/platform-atmelavr/commit/5077ecb2b1c1958fc469e24f2f5c00162924ec1b#diff-0184adedc292a93218525209d37dabe7330a013e6ff8fa46fd6328c5eb4598faR78 it works. Maybe something was missing when MegaCore was updated?

**Comment by GunZi200** [comment]
@valeros @MCUdude I'm thinking that maybe its best to include MegaCore specific MCUs in the example.yml to catch the build errors earlier?

**Comment by GunZi200** [comment]
Ok I looked more into the Github pipeline. There is one MegaCore MCU there and compiles fine. I installed Python 3.7.9 (pipeline uses 3.7.10) and I reinstalled PlatformIO... then the build worked fine. I had Python 3.9.1 before.

Now I've reinstalled 3.9.1 and I am getting this VS Code or SCons error:

![image](https://user-images.githubusercontent.com/8218499/114304499-03eb8900-9ac3-11eb-87b0-3e2186349aef.png)


But then when I try to reproduce this more by reinstalling PIO again without changing the Python installation I get no error. I'm so confused 😅 Will close this for now and reopen if I learn more.

**Comment by GunZi200** [comment]
The issue appeared again on another machine. Adding `SPI` to `platformio.ini` does not work as show in the image below... 
![image](https://user-images.githubusercontent.com/8218499/115207145-d8802400-a0ea-11eb-9881-1325a792e004.png)

**Comment by valeros** [comment]
Have you tried to explicitly specify the `SPI` library as a dependency in `lib_deps`?

**Comment by GunZi200** [comment]
> Have you tried to explicitly specify the `SPI` library as a dependency in `lib_deps`?

Yes I get the same results. You can see the Library Manager trying to install `SPI` in the image above. If I try to install SPI with a fresh platformio install I get a similar yellow text which says SPI library cannot be found for `windows-amd64`. Seems to install OK with the atmelavr framework though.

For reference I did it like this:

```ini
lib_deps =
     SPI
     ; ... more libraries below
```

**Comment by valeros** [comment]
Could you please put together a simple project to reproduce the issue?

**Comment by GunZi200** [comment]
@valeros here: https://github.com/GunZi200/MegaCore-ATmega2560-err

I get this result on my end. I created an Arduino Mega project and only changed the `platformio.ini` file. 

![image](https://user-images.githubusercontent.com/8218499/115278593-29b50580-a135-11eb-976e-7c10f1d19cd0.png)

If you can't reproduce this try one thing I did:

1. Install Python 3.9.1 and add it to PATH
2. delete `.platformio` folder and open vscode.
3. try to build project

I seem to be able to bypass this error when my machine has Python 3.7.9 installed.

**Comment by valeros** [comment]
It's strange, but I cannot reproduce the issue, tried your project with Python 3.9.1 and everything works just fine. Could you please run `pio system info` in your terminal?

**Comment by GunZi200** [comment]
Result from two different computers:

Computer 1:
![image](https://user-images.githubusercontent.com/8218499/115378837-801b5600-a1c0-11eb-93f3-24e8faaaa5a9.png)

Will edit this comment on the other computer to add picture

**Comment by valeros** [comment]
That's super weird. Let's focus on machines where it doesn't work. Do you see the `Wire` library somewhere here  `c:\Users\Gudni\.platformio\packages\framework-arduino-avr-megacore\libraries\`?

**Comment by GunZi200** [comment]
Hmm the folder on the machines where it doesn't work just looks like this: 
![image](https://user-images.githubusercontent.com/8218499/115388022-eefdac80-a1ca-11eb-94a1-e4a4f32bc142.png)

**Comment by valeros** [comment]
Apparently, the `framework-arduino-avr-megacore` package is broken for some reason. Does it always look like that even when you delete the `c:\Users\Gudni\.platformio` folder and try again?

**Comment by GunZi200** [comment]
Hmm I get it to work sometimes but sometimes not. I found one way to trigger the issue when it was working.

I deleted the folder `framework-arduino-avr-megacore` then hit `Clean` and PIO will show me the progress of installing `framework-arduino-avr-megacore` again. I then hit `Build` and issue returns.

**Comment by valeros** [comment]
Do you have an antivirus by any chance?

**Comment by GunZi200** [comment]
No only Windows Defender, its never been a problem for me.

**Comment by valeros** [comment]
Something is messing with PlatformIO packages on your machines and I'm guessing it's `Windows Defender`. Try to create [an exclusion](https://support.microsoft.com/en-us/windows/add-an-exclusion-to-windows-security-811816c0-4dfd-af4a-47e4-c301afe13b26) for `c:\Users\Gudni\.platformio` and then delete this folder to force PlatformIO redownload all packages.

**Comment by GunZi200** [comment]
@valeros I noticed that PlatformIO plugins installs `framework-arduino-avr-megacore` folder when reinstalling the `.platformio` folder. I see not installation progress and it seems to occur while this issue takes places: platformio/platformio-vscode-ide#2506. If I restart VScode while this is happening, that seems to be enough to trigger the issue.

**Comment by GunZi200** [comment]
Looking at the Console... could this be an issue?
![image](https://user-images.githubusercontent.com/8218499/115395160-18bad180-a1d3-11eb-97b7-5ab436600614.png)

**Comment by MCUdude** [comment]
I'm not sure if it's relevant or not, but I was not able to get the example project @GunZi200 provided when platformio.ini looked like this. Identical error to this one: https://github.com/platformio/platform-atmelavr/issues/251#issuecomment-817303556

``` ini
[env:ATmega2560]
platform = atmelavr
board = ATmega2560
framework = arduino

; Libraries
lib_deps =
     Wire
     https://github.com/adafruit/RTClib ; Real-time clock library

lib_ignore = TinyWireM
```

However, if I changed the library dependent to this, it worked:

```ini
[env:ATmega2560]
platform = atmelavr
board = ATmega2560
framework = arduino

; Libraries
lib_deps =
     Wire
     adafruit/RTClib @ 1.13.0

lib_ignore = TinyWireM
```

**Comment by valeros** [comment]
@GunZi200 I believe it has nothing to do with this issue, as your system contains broken framework package.

**Comment by GunZi200** [comment]
I think I have a solution...

Reproduction steps to make it work:

1. Open VScode
2. Find the folder `.platformio\packages\framework-arduino-avr-megacore` and delete it
3. Keep the folder `.platformio\packages` open
4. Now reload VScode window `Ctrl + Shift + P` -> `Developer: Reload Window`
5. `PlatformIO: Rebuilding IntelliSense Index` starts.
6. Notice how the `framework-arduino-avr-megacore` starts with content:

![image](https://user-images.githubusercontent.com/8218499/115396630-b9f65780-a1d4-11eb-90d3-739064c3a859.png)

7. Now wait for `PlatformIO: Rebuilding IntelliSense Index` to finish in VScode an notice how the remaining content in folder `framework-arduino-avr-megacore`  appears ~30 seconds later.

![image](https://user-images.githubusercontent.com/8218499/115396755-ddb99d80-a1d4-11eb-93de-d1b74afd892e.png)

Build from this point works for me on all machines.

If I for some reason reload VScode in step 6 the issue appears and the package is broken. I can consistently reproduce this.

**Comment by valeros** [comment]
@GunZi200 

> If I for some reason reload VScode in step 6 the issue appears and the package is broken. I can consistently reproduce this.

If you do that, you will kill the `platformio-core` process that's running in background, which can lead to unpredictable side effects. In your case, I believe, you interrupted the process when PlatformIO was unpacking the `framework-arduino-avr-megacore` package and leaving it in corrupted state.
